### ISSUE TITLE
Standardization of Changeset Summaries

### ISSUE THREAD

**Issue by SatanshuMishra** [SatanshuMishra/CulinaryCache#7]
Goal: To come up with a standard format to follow when it comes to `changesets`. Add a comment to this issue with a list of things each of us would like to see mentioned as part of the changeset summary. This list will be used to create the standard.

**Comment by SatanshuMishra** [comment]
Things I would like to see in the `changeset`:
- First person language - `This PR...`
- Overarching overview of the changes - `...allows for database manipulation...`

**Comment by VoxelMC** [comment]
Everything you said, plus:
- Large changes that need a bit of an explanation should have usage examples, as well as code examples (if we choose to build an API).
- Configuration additions or changes should have usage examples
- Command name additions or changes should also have usage examples
- A changeset should be made for everything past v1.0. I think that, before then, it doesn't matter as much. Once we get 1.0 working, we should document ALL bug fixes, changes, etc.

LMK what you think.

**Comment by SatanshuMishra** [comment]
> Everything you said, plus:
> 
> * Large changes that need a bit of an explanation should have usage examples, as well as code examples (if we choose to build an API).
> * Configuration additions or changes should have usage examples
> * Command name additions or changes should also have usage examples
> * A changeset should be made for everything past v1.0. I think that, before then, it doesn't matter as much. Once we get 1.0 working, we should document ALL bug fixes, changes, etc.
> 
> LMK what you think.

I agree that usage examples is something we should definitely looks to include. Regarding `changeset`'s before v1.0, I think we should still give them importance not in the context of cataloging every bug or change but to cement the structure we are discussing here. By the time we make that changeset for v0.8 or v0.9, we should have a few well structured changesets under out belt and sorted out any issues.

Other than that, I agree with what you mentioned. Lets give this another day and if no one adds anything to it we can make some templates to see what we like in terms of structure and organization.

Thoughts and comments are welcome :)!

**Comment by VoxelMC** [comment]
> > Everything you said, plus:
> > 
> > * Large changes that need a bit of an explanation should have usage examples, as well as code examples (if we choose to build an API).
> > * Configuration additions or changes should have usage examples
> > * Command name additions or changes should also have usage examples
> > * A changeset should be made for everything past v1.0. I think that, before then, it doesn't matter as much. Once we get 1.0 working, we should document ALL bug fixes, changes, etc.
> > 
> > LMK what you think.
> 
> I agree that usage examples is something we should definitely looks to include. Regarding `changeset`'s before v1.0, I think we should still give them importance not in the context of cataloging every bug or change but to cement the structure we are discussing here. By the time we make that changeset for v0.8 or v0.9, we should have a few well structured changesets under out belt and sorted out any issues.
> 
> Other than that, I agree with what you mentioned. Lets give this another day and if no one adds anything to it we can make some templates to see what we like in terms of structure and organization.
> 
> Thoughts and comments are welcome :)!

Yes, agreed on the pre-v1.0 things. I really just meant not having changesets like `added parser` or something dumb and big like that.
